pio-dist/setup.sh

```diff
@@ -5,18 +5,26 @@ region=`curl -s http://169.254.169.254/latest/meta-data/placement/availability-z
 
 stagetag=`aws ec2 describe-tags --filters "Name=resource-id,Values=$instanceid" "Name=resource-type,Values=instance" "Name=key,Values=Stage" --region $region | grep -oP "(?<=\"Value\": \")[^\"]+"`
 
-HADOOP_MASTER=namenode.hadoop.recommendations.$stagetag.internal.
-ELASTICSEARCH_HOSTS=nodes.elasticsearch.recommendations.$stagetag.internal.
+HADOOP_MASTER=master-0.recommendations.prod.internal.
+
+ES_HOSTS_TMP=`mktemp`
+aws ec2 describe-instances --region eu-west-1 --filters "Name=tag:App,Values=elasticsearch" "Name=tag:Stage,Values=$stagetag" \
+    --query "Reservations[].Instances[].{ Dns: PrivateDnsName}" \
+    | grep -oP "(?<=\"Dns\": \")[^\"]+" > "$ES_HOSTS_TMP"
+
+ES_HOSTS=`paste -sd "," ${ES_HOSTS_TMP}`
+ES_PORTS=`cat ${ES_HOSTS_TMP} | sed 's/.*/9300/' | paste -sd "," -`
```

Comment:
You don't need this to be a comma separated list of 9300.  Just one 9300 will do

See https://github.com/PredictionIO/PredictionIO/blob/master/data/src/main/scala/io/prediction/data/storage/elasticsearch/StorageClient.scala#L38